File: server/api/v1/shop/shop_brand.go
package shop

import (
	"github.com/flipped-aurora/gin-vue-admin/server/global"
	"github.com/flipped-aurora/gin-vue-admin/server/model/common/response"
	"github.com/flipped-aurora/gin-vue-admin/server/model/shop"
	shopReq "github.com/flipped-aurora/gin-vue-admin/server/model/shop/request"
	"github.com/flipped-aurora/gin-vue-admin/server/service"
	"github.com/gin-gonic/gin"
	"go.uber.org/zap"
)

type ShopBrandApi struct{}

var shopBrandService = service.ServiceGroupApp.ShopServiceGroup.ShopBrandService

func (a *ShopBrandApi) CreateShopBrand(c *gin.Context) {
	var brand shop.ShopBrand
	if err := c.ShouldBindJSON(&brand); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	if err := shopBrandService.CreateShopBrand(&brand); err != nil {
		global.GVA_LOG.Error("创建失败!", zap.Error(err))
		response.FailWithMessage("创建失败", c)
	} else {
		response.OkWithMessage("创建成功", c)
	}
}

func (a *ShopBrandApi) DeleteShopBrand(c *gin.Context) {
	ID := c.Query("ID")
	if ID == "" {
		response.FailWithMessage("参数错误", c)
		return
	}
	if err := shopBrandService.DeleteShopBrand(ID); err != nil {
		global.GVA_LOG.Error("删除失败!", zap.Error(err))
		response.FailWithMessage("删除失败", c)
	} else {
		response.OkWithMessage("删除成功", c)
	}
}

func (a *ShopBrandApi) UpdateShopBrand(c *gin.Context) {
	var brand shop.ShopBrand
	if err := c.ShouldBindJSON(&brand); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	if err := shopBrandService.UpdateShopBrand(brand); err != nil {
		global.GVA_LOG.Error("更新失败!", zap.Error(err))
		response.FailWithMessage("更新失败", c)
	} else {
		response.OkWithMessage("更新成功", c)
	}
}

func (a *ShopBrandApi) FindShopBrand(c *gin.Context) {
	ID := c.Query("ID")
	if brand, err := shopBrandService.GetShopBrand(ID); err != nil {
		global.GVA_LOG.Error("查询失败!", zap.Error(err))
		response.FailWithMessage("查询失败", c)
	} else {
		response.OkWithData(gin.H{"data": brand}, c)
	}
}

func (a *ShopBrandApi) GetShopBrandList(c *gin.Context) {
	var pageInfo shopReq.ShopBrandSearch
	if err := c.ShouldBindQuery(&pageInfo); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	if list, total, err := shopBrandService.GetShopBrandList(pageInfo); err != nil {
		global.GVA_LOG.Error("获取失败!", zap.Error(err))
		response.FailWithMessage("获取失败", c)
	} else {
		response.OkWithDetailed(response.PageResult{
			List:     list,
			Total:    total,
			Page:     pageInfo.Page,
			PageSize: pageInfo.PageSize,
		}, "获取成功", c)
	}
}

func (a *ShopBrandApi) GetAllBrands(c *gin.Context) {
	if list, err := shopBrandService.GetAllBrands(); err != nil {
		global.GVA_LOG.Error("获取失败!", zap.Error(err))
		response.FailWithMessage("获取失败", c)
	} else {
		response.OkWithData(gin.H{"list": list}, c)
	}
}

File: web/src/api/shopBrand.js
import service from '@/utils/request'

// @Tags ShopBrand
// @Summary 创建品牌
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.ShopBrand true "创建品牌"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /shopBrand/createShopBrand [post]
export const createShopBrand = (data) => {
  return service({
    url: '/shopBrand/createShopBrand',
    method: 'post',
    data
  })
}

// @Tags ShopBrand
// @Summary 删除品牌
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.ShopBrand true "删除品牌"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /shopBrand/deleteShopBrand [delete]
export const deleteShopBrand = (params) => {
  return service({
    url: '/shopBrand/deleteShopBrand',
    method: 'delete',
    params
  })
}

// @Tags ShopBrand
// @Summary 更新品牌
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.ShopBrand true "更新品牌"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /shopBrand/updateShopBrand [put]
export const updateShopBrand = (data) => {
  return service({
    url: '/shopBrand/updateShopBrand',
    method: 'put',
    data
  })
}

// @Tags ShopBrand
// @Summary 用id查询品牌
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query model.ShopBrand true "用id查询品牌"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /shopBrand/findShopBrand [get]
export const findShopBrand = (params) => {
  return service({
    url: '/shopBrand/findShopBrand',
    method: 'get',
    params
  })
}

// @Tags ShopBrand
// @Summary 分页获取品牌列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取品牌列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /shopBrand/getShopBrandList [get]
export const getShopBrandList = (params) => {
  return service({
    url: '/shopBrand/getShopBrandList',
    method: 'get',
    params
  })
}

// @Tags ShopBrand
// @Summary 获取全部品牌
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "获取全部品牌"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /shopBrand/getAllBrands [get]
export const getAllBrands = (params) => {
  return service({
    url: '/shopBrand/getAllBrands',
    method: 'get',
    params
  })
}
